activemodel-serializers-xml.gemspec

```diff
@@ -15,11 +15,11 @@ Gem::Specification.new do |spec|
 
   spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   spec.require_paths = ["lib"]
-  spec.add_dependency "activesupport", "> 5.x"
-  spec.add_dependency "activemodel", "> 5.x"
+  spec.add_dependency "activesupport", "> 5"
```

Comment:
This will not allow 5.0.0.beta1. This is why we had a letter there.

Reply:
I don't understand. "> 5.x" intended to allow 5.0.0.beta1?

Reply:
Yes, any pre-release of the 5 series.

Reply:
Sorry, my change was incorrect (at least not what I meant). Technically "> 5.x" is equivalent to ">= 5". That's the change I wanted to make. But none of these requirements allow prereleases, I think? Or at least, I don't think they should?

Reply:
> Yes, any pre-release of the 5 series.

Ok, then it's great that I brought up this issue. Can you explain the reasoning behind your expectation? What you expect _is_ how rubygems currently works, but in my opinion it is due to a bug. If I had to allow prereleases higher than or equal to 5.0.0.beta1, I would say `>= 5.0.0.beta1`.

Reply:
Because we want that people trying the 5.0.0.beta1, even the alpha (master branch) can use this gem. I don't mind changing this to 5.a if that looks more correct, but we need to allow people in any pre release version to use this gem.

Reply:
If you don't know in advance the names of the prereleases you want to allow, but you want to allow them all, rubygems recommends using `.a`, because it's the least possible prerelease, lexicographically speaking.

> If you want to specify a version restriction that includes both prereleases
> and regular releases of the 1.x series this is the best way:
>
> s.add_dependency 'example', '>= 1.0.0.a', '< 2.0.0'

Reference: https://github.com/rubygems/rubygems/blob/2da79d40485d8b15e79de146936f0daf300e6470/lib/rubygems/version.rb#L26-L30

Reply:
> Because we want that people trying the 5.0.0.beta1, even the alpha (master branch) can use this gem. I don't mind changing this to 5.a if that looks more correct, but we need to allow people in any pre release version to use this gem.

Yes, sure, I understand the motivations for allowing prereleases :+1:. What I meant is why you used "> 5.x" in the beginning. I guess just because you tried it and it worked?

Reply:
In any case, yeah, the more correct thing in my opinion would be either "> 5.a" or ">= 5.0.0.beta1". I'm just considering whether we should fix this bug and looking for gems depending on the incorrect behavior found this one and activemodel-serializers-xml.

Reply:
the `x` is because I learned that Rubygems only require any letter to allow pre releases.

Reply:
I see, yeah, I learnt the same thing I guess because I saw it somewhere, tried it and it worked. But I recently found out that it works like that only due to a bug. The expectation from reading the rubygems documentation is that the prerelease part should work for comparison just like the release part, except alphabetically. For example,

1.0.0.a < 1.0.0.alpha < 1.0.0.beta1 < 1.0.0.beta2 < 1.0.0.rc1 < 1.0.0.rc2 < 1.0.0.x < 1.0.0.zeitwerk